eclass/kernel-install.eclass

```diff
@@ -442,6 +595,26 @@ kernel-install_pkg_preinst() {
 	fi
 }
 
+# @FUNCTION: kernel-install_extract_kernel
+# @USAGE: <kernel image>
+# @DESCRIPTION:
+# Extracts kernel image from UKI, removes UKI if it is not wanted
+kernel-install_extract_kernel() {
+	[[ ${#} -eq 1 ]] || die "${FUNCNAME}: invalid arguments"
+	local image_path=${1}
+	local uki="${image_path%/*}/uki.efi"
+
+	$(tc-getOBJCOPY) -O binary -j.linux "${uki}" "${image_path}" ||
+		die "Failed to extract kernel image"
+
+	if ! use generic-uki; then
+		# If our gpkg was built with USE=generic-uki, but we don't want
+		# to use it, remove it, otherwise installkernel will pick it
+		# up automatically.
+		rm "${uki}" || die "Failed to remove UKI"
+	fi
```

Comment:
1. Remove this part, make the function "pure" extraction.
2. Make `.linux`, `.initrd` an argument (default to `.linux`), then you can reuse it for both extractions.